watchtower/wtclient/client.go

```diff
@@ -1042,6 +1069,76 @@ func (c *client) initActiveQueue(s *ClientSession,
 	return sq
 }
 
+// deactivateTower sets the given tower's status to inactive so that it
+// is not considered for session negotiation. Its sessions will also not
+// be used while the tower is inactive.
+func (c *client) deactivateTower(id wtdb.TowerID,
+	pubKey *btcec.PublicKey) error {
+
+	errChan := make(chan error, 1)
+
+	select {
+	case c.deactivateTowers <- &deactivateTowerMsg{
+		id:      id,
+		pubKey:  pubKey,
+		errChan: errChan,
+	}:
+	case <-c.pipeline.quit:
+		return ErrClientExiting
+	}
+
+	select {
+	case err := <-errChan:
+		return err
+	case <-c.pipeline.quit:
+		return ErrClientExiting
+	}
+}
+
+// handleDeactivateTower handles a request to deactivate a tower. We will remove
+// it from the in-memory candidate set, and we will also stop any active
+// sessions we have with this tower.
+func (c *client) handleDeactivateTower(msg *deactivateTowerMsg) error {
+	// Remove the tower from our in-memory candidate set so that it is not
+	// used for any new session negotiations.
+	err := c.candidateTowers.RemoveCandidate(msg.id, nil)
+	if err != nil {
+		return err
+	}
+
+	pubKey := msg.pubKey.SerializeCompressed()
+	sessions, err := c.cfg.DB.ListClientSessions(&msg.id)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve sessions for tower %x: "+
+			"%v", pubKey, err)
+	}
+
+	// Iterate over all the sessions we have for this tower and remove them
+	// from our candidate set and also from our set of started, active
+	// sessions.
+	for sessionID := range sessions {
+		delete(c.candidateSessions, sessionID)
+
+		err = c.activeSessions.StopAndRemove(sessionID, false)
+		if err != nil {
+			return fmt.Errorf("could not stop session %s: %w",
+				sessionID, err)
+		}
+	}
+
+	// If our active session queue corresponds to the stale tower, we'll
+	// proceed to negotiate a new one.
+	if c.sessionQueue != nil {
+		towerKey := c.sessionQueue.tower.IdentityKey
+
+		if bytes.Equal(pubKey, towerKey.SerializeCompressed()) {
+			c.sessionQueue = nil
```

Comment:
do you mean just extracting the `c.sessionQueue = nil` part into its own method? if so - i think we might want to make it clear what we are doing here since setting things to nil is kind-of an anti-pattern so dont want to accidentally cause a nil panic. ideally we should actually have a long living object that we can call "Get/SetNil" etc on so that we dont run the risk of nil panic with this. But think that refactor should be in a different pr

Reply:
cool, just wanted to mention it, yes seems great for another PR